### ISSUE TITLE
ionic redirect URL

### ISSUE THREAD

**Issue by sraveh** [sahat/satellizer#675]
Hi,

I'm trying to use the google and twitter login with ionic. I follow the the setup direction but the redirect does not work.
My application is running on port 8100, when I change the redirect to http://localhost:8100 the library does complete the authentication process. Do I need to process the redirect or does the library handle it directly

**Comment by sahat** [comment]
If your app is running on port 8100 then you need to set redirect to http://localhost:8100 in your app where you configure Satellizer and in the app settings of an OAuth provider.

On a related note, does Ionic Framework run an app on `http://localhost:8100` by default? I thought it was just `http://localhost`. That's why `redirectUri` was configured to use that URL. https://github.com/sahat/satellizer/blob/master/examples/ionic/www/js/app.js#L28

**Comment by sraveh** [comment]
Yes, I'm running via ionic serve. Default port is 8100. When I change the redirect to http://localhost:8100 I can see the callback URL in the popup window but instead of continue the authentication process, the popup window just loads my application. 
Does the library support both mobile app and mobile web?. should it work with Ionic Serve

Simon

**Comment by sahat** [comment]
Yes it did at one point. I just updated the Ionic example it now uses a commonConfig object to set popup options for the Cordova browser as well as different redirectUri when running an iOS/Android Emulator.

```js
  if (ionic.Platform.isIOS() || ionic.Platform.isAndroid()) {
      $authProvider.cordova = true;
      commonConfig.redirectUri = 'http://localhost/';
    }
  // otherwise, `redirectUri` to whatever URL your app is running in, e.g. http://localhost:8100.
```

**Comment by sahat** [comment]
@sraveh Could you verify that it works now and close this issue?

**Comment by sraveh** [comment]
Hi,

This fixed the issue of redirecting correctly back to the application but I'm still seeing the original issue. The oauth authentication flow is not finish and the ionic application is being loaded in the authentication popup window. 

Thanks,
Simon

**Comment by sahat** [comment]
@sraveh I have double checked everything but still could not reproduce your error. Can you copy and paste your popup's URL here? Typically, if it has a `?code=` query parameter, Satellizer will detect it and close the popup. Similarly if it contains `?error=` it will properly handle that error and close the popup. Your popup either contains unsupported parameter or there is something else causing this problem.

Also, which OAuth provider are you using?

**Comment by alexandrupetrescu94** [comment]
Hi,

I am having the same problem as @sraveh . I am authenticating with google,and when I redirect, it redirects me to the popup window with

*http://localhost:5000/?code=4/{token}&authuser=0&session_state={session_state}&prompt=constent#login*

OAuth2, and when I am consoling here it doesn't work anymore:
satellizer.js, line ~525

```javascript
return openPopup.then(function(oauthData) { console.log('working?');});
```

Thank you!
